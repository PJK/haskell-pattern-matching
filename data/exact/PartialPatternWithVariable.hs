module PartialPatternWithVariable where

data MyTrinary = A | B | C

data MyMaybe = Sum MyTrinary | Nope

partialExtract :: MyMaybe -> MyMaybe -> MyTrinary
partialExtract (Sum C) (Sum A) = A
partialExtract (Sum x)  y      = B
partialExtract  Nope   (Sum C) = B

