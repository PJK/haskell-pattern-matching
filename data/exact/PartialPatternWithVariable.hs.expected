{
    "tag": "AnalysisSuccess",
    "contents": [
        [
            [
                "partialExtract",
                {
                    "tag": "NonExhaustive",
                    "contents": [
                        [
                            [
                                {
                                    "tag": "ConstructorPattern",
                                    "contents": [
                                        "Nope",
                                        []
                                    ]
                                },
                                {
                                    "tag": "ConstructorPattern",
                                    "contents": [
                                        "Sum",
                                        [
                                            {
                                                "tag": "ConstructorPattern",
                                                "contents": [
                                                    "A",
                                                    []
                                                ]
                                            }
                                        ]
                                    ]
                                }
                            ],
                            ""
                        ],
                        [
                            [
                                {
                                    "tag": "ConstructorPattern",
                                    "contents": [
                                        "Nope",
                                        []
                                    ]
                                },
                                {
                                    "tag": "ConstructorPattern",
                                    "contents": [
                                        "Sum",
                                        [
                                            {
                                                "tag": "ConstructorPattern",
                                                "contents": [
                                                    "B",
                                                    []
                                                ]
                                            }
                                        ]
                                    ]
                                }
                            ],
                            ""
                        ],
                        [
                            [
                                {
                                    "tag": "ConstructorPattern",
                                    "contents": [
                                        "Nope",
                                        []
                                    ]
                                },
                                {
                                    "tag": "ConstructorPattern",
                                    "contents": [
                                        "Nope",
                                        []
                                    ]
                                }
                            ],
                            ""
                        ]
                    ]
                }
            ]
        ],
        [
            [
                "partialExtract",
                [
                    [
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "VariablePattern",
                                "contents": "~a"
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "Sum",
                                    [
                                        {
                                            "tag": "VariablePattern",
                                            "contents": "~i"
                                        }
                                    ]
                                ]
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "Sum",
                                    [
                                        {
                                            "tag": "VariablePattern",
                                            "contents": "~i"
                                        }
                                    ]
                                ]
                            },
                            {
                                "tag": "WildcardPattern",
                                "contents": []
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "Sum",
                                    [
                                        {
                                            "tag": "ConstructorPattern",
                                            "contents": [
                                                "C",
                                                []
                                            ]
                                        }
                                    ]
                                ]
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "Sum",
                                    [
                                        {
                                            "tag": "ConstructorPattern",
                                            "contents": [
                                                "C",
                                                []
                                            ]
                                        }
                                    ]
                                ]
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "Sum",
                                    [
                                        {
                                            "tag": "ConstructorPattern",
                                            "contents": [
                                                "C",
                                                []
                                            ]
                                        }
                                    ]
                                ]
                            },
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "Sum",
                                    [
                                        {
                                            "tag": "VariablePattern",
                                            "contents": "~j"
                                        }
                                    ]
                                ]
                            }
                        ],
                        [
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "Sum",
                                    [
                                        {
                                            "tag": "ConstructorPattern",
                                            "contents": [
                                                "C",
                                                []
                                            ]
                                        }
                                    ]
                                ]
                            },
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "Sum",
                                    [
                                        {
                                            "tag": "VariablePattern",
                                            "contents": "~j"
                                        }
                                    ]
                                ]
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "Nope",
                                    []
                                ]
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ],
                        [
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "Nope",
                                    []
                                ]
                            },
                            {
                                "tag": "VariablePattern",
                                "contents": "~b"
                            }
                        ]
                    ],
                    [
                        [
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "Nope",
                                    []
                                ]
                            },
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "Sum",
                                    [
                                        {
                                            "tag": "VariablePattern",
                                            "contents": "~n"
                                        }
                                    ]
                                ]
                            }
                        ],
                        [
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "Nope",
                                    []
                                ]
                            },
                            {
                                "tag": "ConstructorPattern",
                                "contents": [
                                    "Sum",
                                    [
                                        {
                                            "tag": "VariablePattern",
                                            "contents": "~n"
                                        }
                                    ]
                                ]
                            }
                        ]
                    ]
                ]
            ]
        ]
    ]
}
